File: scripts/update_invite_translations.py
import json
import os

translations = {
  'ar': 'بدعوة فقط',
  'de': 'NUR AUF EINLADUNG',
  'en': 'INVITE-ONLY',
  'es': 'SÓLO POR INVITACIÓN',
  'fr': 'SUR INVITATION UNIQUEMENT',
  'it': 'SOLO SU INVITO',
  'nl': 'UITSLUITEND OP UITNODIGING',
  'pt': 'APENAS CONVIDADOS',
  'ru': 'ТОЛЬКО ПО ПРИГЛАШЕНИЮ',
  'zh': '仅限邀请'
}

locales_dir = os.path.join('src', 'i18n', 'locales')

if os.path.exists(locales_dir):
  for lang in os.listdir(locales_dir):
    lang_dir = os.path.join(locales_dir, lang)
    if not os.path.isdir(lang_dir): continue
    target_text = translations.get(lang, translations['en'])

    for file in os.listdir(lang_dir):
      if not file.endswith('.json'): continue
      file_path = os.path.join(lang_dir, file)
      try:
        with open(file_path, 'r', encoding='utf-8') as f:
          data = json.load(f)
        
        modified = False
        if 'inviteOnly' in data:
          data['inviteOnly'] = target_text
          modified = True
          
        if 'signup' in data and 'inviteOnly' in data['signup']:
          data['signup']['inviteOnly'] = target_text
          modified = True
          
        if modified:
          with open(file_path, 'w', encoding='utf-8') as f:
            json.dump(data, f, indent=2, ensure_ascii=False)
            f.write('\n')
          print(f"Updated {lang}/{file} to '{target_text}'")
      except Exception as e:
        print(f"Error with {file_path}: {e}")

File: scripts/update_invite_translations.js
const fs = require('fs');
const path = require('path');

const translations = {
  ar: 'بدعوة فقط',
  de: 'NUR AUF EINLADUNG',
  en: 'INVITE-ONLY',
  es: 'SÓLO POR INVITACIÓN',
  fr: 'SUR INVITATION UNIQUEMENT',
  it: 'SOLO SU INVITO',
  nl: 'UITSLUITEND OP UITNODIGING',
  pt: 'APENAS CONVIDADOS',
  ru: 'ТОЛЬКО ПО ПРИГЛАШЕНИЮ',
  zh: '仅限邀请'
};

const localesDir = path.join(__dirname, '..', 'src', 'i18n', 'locales');

if (fs.existsSync(localesDir)) {
  const langs = fs.readdirSync(localesDir);
  for (const lang of langs) {
    const langDir = path.join(localesDir, lang);
    if (!fs.statSync(langDir).isDirectory()) continue;
    
    const targetText = translations[lang] || translations.en;
    
    // update auth.json or common.json
    for (const file of fs.readdirSync(langDir)) {
      if (!file.endsWith('.json')) continue;
      
      const filePath = path.join(langDir, file);
      try {
        const content = fs.readFileSync(filePath, 'utf8');
        const data = JSON.parse(content);
        
        let modified = false;
        
        // 1. Root level inviteOnly
        if (data.inviteOnly !== undefined) {
          data.inviteOnly = targetText;
          modified = true;
        }
        
        // 2. if signup.inviteOnly
        if (data.signup && data.signup.inviteOnly !== undefined) {
          data.signup.inviteOnly = targetText;
          modified = true;
        }
        
        if (modified) {
          fs.writeFileSync(filePath, JSON.stringify(data, null, 2) + '\n', 'utf8');
          console.log(`Updated ${lang}/${file} to "${targetText}"`);
        }
      } catch (err) {
        console.error(`Error in ${lang}/${file}: ${err.message}`);
      }
    }
  }
} else {
  console.error("Locales directory not found", localesDir);
}
